src/ray/raylet/local_object_manager.cc

```diff
@@ -41,22 +41,17 @@ void LocalObjectManager::PinObjects(const std::vector<ObjectID> &object_ids,
 void LocalObjectManager::WaitForObjectFree(const rpc::Address &owner_address,
                                            const std::vector<ObjectID> &object_ids) {
   for (const auto &object_id : object_ids) {
-    // Send a long-running RPC request to the owner for each object. When we get a
-    // response or the RPC fails (due to the owner crashing), unpin the object.
-    // TODO(edoakes): we should be batching these requests instead of sending one per
-    // pinned object.
-    rpc::WaitForObjectEvictionRequest wait_request;
-    wait_request.set_object_id(object_id.Binary());
-    wait_request.set_intended_worker_id(owner_address.worker_id());
-    auto owner_client = owner_client_pool_.GetOrConnect(owner_address);
-    owner_client->WaitForObjectEviction(
-        wait_request,
-        [this, object_id](Status status, const rpc::WaitForObjectEvictionReply &reply) {
-          if (!status.ok()) {
-            RAY_LOG(DEBUG) << "Worker failed. Unpinning object " << object_id;
-          }
-          ReleaseFreedObject(object_id);
-        });
+    // Callback that is invoked when the owner publishes the object to evict.
+    auto subscription_callback = [this, owner_address](const ObjectID &object_id) {
+      ReleaseFreedObject(object_id);
+      core_worker_pubsub_client_->UnsubscribeObject(owner_address, object_id);
+    };
+    // Callback that is invoked when the owner of the object id is dead.
+    auto owner_dead_callback = [this](const ObjectID &object_id) {
+      ReleaseFreedObject(object_id);
+    };
+    core_worker_pubsub_client_->SubcribeObject(
+        owner_address, object_id, subscription_callback, owner_dead_callback);
```

Comment:
Not necessary for this PR, but we should have a batch implementation of this as well so you can subscribe to multiple objects at once with a single subscribe RPC. I'm guessing that the proto level request would be batch (`repeated bytes object_ids`).

Reply:
Are you saying we should include this in the long polling request? I think it is fine not to do that because there was no issue from this that significantly impacts the performance. I can add a TODO though.

Reply:
Ah my bad. I actually just saw `WaitForObjectFree` receives multiple object ids as input. Anyway I will add a TODO here (and maybe touch it later).